src/openrct2-ui/windows/Footpath.cpp

```diff
@@ -563,26 +558,52 @@ static void window_footpath_invalidate(rct_window* w)
     // Press / unpress footpath and queue type buttons
     w->pressed_widgets &= ~(1ULL << WIDX_FOOTPATH_TYPE);
     w->pressed_widgets &= ~(1ULL << WIDX_QUEUELINE_TYPE);
-    w->pressed_widgets |= _footpathSelectedType == SELECTED_PATH_TYPE_NORMAL ? (1ULL << WIDX_FOOTPATH_TYPE)
-                                                                             : (1ULL << WIDX_QUEUELINE_TYPE);
+    w->pressed_widgets |= gFootpathSelection.IsQueueSelected ? (1ULL << WIDX_QUEUELINE_TYPE) : (1ULL << WIDX_FOOTPATH_TYPE);
 
     // Enable / disable construct button
     window_footpath_widgets[WIDX_CONSTRUCT].type = _footpathConstructionMode == PATH_CONSTRUCTION_MODE_BRIDGE_OR_TUNNEL
         ? WindowWidgetType::ImgBtn
         : WindowWidgetType::Empty;
 
-    // Set footpath and queue type button images
-    auto pathImage = static_cast<uint32_t>(SPR_NONE);
-    auto queueImage = static_cast<uint32_t>(SPR_NONE);
-    auto pathEntry = get_path_surface_entry(gFootpathSelectedId);
-    if (pathEntry != nullptr)
+    if (gFootpathSelection.LegacyPath == OBJECT_ENTRY_INDEX_NULL)
     {
-        pathImage = pathEntry->preview;
-        // Editor-only paths might lack a queue image
-        queueImage = (pathEntry->flags & FOOTPATH_ENTRY_FLAG_SHOW_ONLY_IN_SCENARIO_EDITOR) ? pathImage : pathImage + 1;
```

Comment:
This change in image based on if only in scenario editor has been removed. Im not really sure its possible to have a pathEntry that has it set at this point as it could never be selected from the dropdown but just checking that this was meant to change.

Reply:
The logic has now been moved to object loading (merged in a separate PR).